### PR TITLE
Add platform test for kdump

### PR DESCRIPTION
**Pull request by amnsinghal**
Summary:
Add test_kdump.py to platform_tests.

### Type of change
- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [X] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [X] 202205

#### What is the motivation for this PR?
Add automation test for kdump functionality.

#### How did you do it?
Induce kernel panic thru sysrq-trigger.

#### How did you verify/test it?
For platform with kdump enabled, trigger kernel panic, and check for reboot-cause on subsequent bootup.
Validated on Cisco-8000 platforms thru internal sonic-mgmt test.

image

#### Any platform specific information?
N/A

#### Supported testbed topology if it's a new test case?
Any platform with kdump enabled, else test will be skipped.

### CODE REVIEW

**Comment by amnsinghal** [comment]
@prgeor @lguohan @abdosi Pls help review this new testcase for kdump, thanks.

**Comment by amnsinghal** [comment]
@prgeor Can you pls help check this Elastictest failure, doesn't seem related to my change, thanks.

**Comment by amnsinghal** [comment]
@prgeor Can you pls help review and check this Elastictest failure, doesn't seem related to my change, thanks.

File: tests/platform_tests/test_kdump.py
Comment (bpar9) [inline tests/platform_tests/test_kdump.py:65]: do we support the entire behavior in this document - https://github.com/sonic-net/SONiC/blob/master/doc/kdump/SONiC-kdump.md . If so, then why skip the test if disabled, cant we enable kdump with CLI and then proceed with the test -
root@croc-xx36-dut:/home/cisco# config kdump ?
Usage: config kdump [OPTIONS] COMMAND [ARGS]...

Configure the KDUMP mechanism

Options:
-h, -?, --help Show this message and exit.

Commands:
disable Disable the KDUMP mechanism
enable Enable the KDUMP mechanism
memory Configure the memory for KDUMP mechanism
num_dumps Configure the maximum dump files of KDUMP mechanism

File: tests/platform_tests/test_kdump.py
Comment (bpar9) [inline tests/platform_tests/test_kdump.py:87]: can you use the common proc - check_reboot_cause from common/reboot.py

**Review by bpar9 — CHANGES_REQUESTED** [review]

**Comment by bpar9** [comment]
@yxieca there is an old PR which has a superset of TCs then what is present in this PR. That PR has been reviewed but not merged for past couple of years- https://github.com/sonic-net/sonic-mgmt/pull/3562 . Is there some reason why community has not merged the old PR

**Comment by yxieca** [comment]
> @yxieca there is an old PR which has a superset of TCs then what is present in this PR. That PR has been reviewed but not merged for past couple of years- #3562 . Is there some reason why community has not merged the old PR

Good question. Thanks for looking up the history. I believe the old PR #3562 has unresolve review comments. That was the reason for leaving it open.

File: tests/platform_tests/test_kdump.py
Comment (amnsinghal) [reply tests/platform_tests/test_kdump.py:65]: This test will run only if kdump is already enabled, as we are not sure if other platforms support it.

File: tests/platform_tests/test_kdump.py
Comment (amnsinghal) [reply tests/platform_tests/test_kdump.py:87]: Should I add 'Kernel Panic' to ```reboot_ctrl_dict``` ?

**Comment by bpar9** [comment]
> > @yxieca there is an old PR which has a superset of TCs then what is present in this PR. That PR has been reviewed but not merged for past couple of years- #3562 . Is there some reason why community has not merged the old PR
> 
> Good question. Thanks for looking up the history. I believe the old PR #3562 has unresolve review comments. That was the reason for leaving it open.

thanks @yxieca can the reply. So what is your guidance on this PR-10047? it has a subset of the TCs of #3562 which are currently applicable for our platform. So do we wait for 3562 to be merged or can we go ahead and push this PR in. Please let us know, so that we can proceed accordingly. Thanks

**Comment by yxieca** [comment]
> > > @yxieca there is an old PR which has a superset of TCs then what is present in this PR. That PR has been reviewed but not merged for past couple of years- #3562 . Is there some reason why community has not merged the old PR
> > 
> > 
> > Good question. Thanks for looking up the history. I believe the old PR #3562 has unresolve review comments. That was the reason for leaving it open.
> 
> thanks @yxieca for the reply. So what is your guidance on this PR-10047? it has a subset of the TCs of #3562 which are currently applicable for our platform. So do we wait for 3562 to be merged or can we go ahead and push this PR in. Please let us know, so that we can proceed accordingly. Thanks

We can work on this PR and get it merged. At this point. this PR contains blocking comments. Please resolve.

File: tests/platform_tests/test_kdump.py
Comment (amnsinghal) [reply tests/platform_tests/test_kdump.py]: Updated.

**Comment by bpar9** [comment]
> > > > @yxieca there is an old PR which has a superset of TCs then what is present in this PR. That PR has been reviewed but not merged for past couple of years- #3562 . Is there some reason why community has not merged the old PR
> > > 
> > > 
> > > Good question. Thanks for looking up the history. I believe the old PR #3562 has unresolve review comments. That was the reason for leaving it open.
> > 
> > 
> > thanks @yxieca for the reply. So what is your guidance on this PR-10047? it has a subset of the TCs of #3562 which are currently applicable for our platform. So do we wait for 3562 to be merged or can we go ahead and push this PR in. Please let us know, so that we can proceed accordingly. Thanks
> 
> We can work on this PR and get it merged. At this point. this PR contains blocking comments. Please resolve.

thanks @yxieca , Aman will work on addressing  the comments and proceed.

**Comment by amnsinghal** [comment]
> > > > > @yxieca there is an old PR which has a superset of TCs then what is present in this PR. That PR has been reviewed but not merged for past couple of years- #3562 . Is there some reason why community has not merged the old PR
> > > > 
> > > > 
> > > > Good question. Thanks for looking up the history. I believe the old PR #3562 has unresolve review comments. That was the reason for leaving it open.
> > > 
> > > 
> > > thanks @yxieca for the reply. So what is your guidance on this PR-10047? it has a subset of the TCs of #3562 which are currently applicable for our platform. So do we wait for 3562 to be merged or can we go ahead and push this PR in. Please let us know, so that we can proceed accordingly. Thanks
> > 
> > 
> > We can work on this PR and get it merged. At this point. this PR contains blocking comments. Please resolve.
> 
> thanks @yxieca , Aman will work on addressing the comments and proceed.

Thanks for confirming @yxieca and @bpar9, I have resolved the comments and update the PR, pls review and approve/merge.

**Comment by amnsinghal** [comment]
@yxieca All checks have passed, except Semgrep is still waiting for status to be reported, can you pls check this. Thanks!

**Comment by yxieca** [comment]
@prgeor can you review this test?

**Review by bpar9 — APPROVED** [review]

**Comment by amnsinghal** [comment]
@prgeor Can you pls review and approve/merge, if no further suggestions, thanks!

File: tests/platform_tests/test_kdump.py
Comment (prgeor) [inline tests/platform_tests/test_kdump.py:27]: @amnsinghal is this test supposed to run only on supervisor?

File: tests/platform_tests/test_kdump.py
Comment (prgeor) [inline tests/platform_tests/test_kdump.py:67]: @amnsinghal why not trigger panic via `echo c > /proc/sysrq-trigger`

File: tests/platform_tests/test_kdump.py
Comment (bmridul) [reply tests/platform_tests/test_kdump.py:27]: No, the test runs on both LCs and RPs. For RPs, there is an additional check to ensure that LCs come up healthy after RP goes down and come back up. (In SONiC distributed chassis, LCs go down as well if RP goes down - Ref: https://github.com/sonic-net/SONiC/pull/945#discussion_r828304573)

File: tests/platform_tests/test_kdump.py
Comment (bmridul) [reply tests/platform_tests/test_kdump.py:67]: That is indeed what is being done with "REBOOT_TYPE_KERNEL_PANIC". See changes in reboot.py above.

**Comment by amulyan7** [comment]
@prgeor if no further comments, can you please approve/merge, thanks!

**Comment by bpar9** [comment]
@prgeor , @yxieca , as we addressed all the comments, can we go ahead and merge this? thanks

**Comment by shivuv** [comment]
> @prgeor , @yxieca , as we addressed all the comments, can we go ahead and merge this? thanks

Team: Can you please merge this PR ?

**Comment by shivuv** [comment]
@prgeor  @yxieca  @gechiang @abdosi : Please help to merge this PR into master and cherry pick this into 202305 and 202311 branches.

**Comment by abdosi** [comment]
@wangxin please help with cherry-pick into 202205/202305.

**Pull request merged**
